spec/support/acceptance/shared_contexts.rb

```diff
@@ -26,3 +26,19 @@
     IncidentHelpers.delete_incidents(query)
   end
 end
+
+RSpec.shared_context 'corrective change' do |noop = false|
+  before(:each) do
+    # For a change to be considered corrective, the resource needs to be managed and in
+    # the correct state at least one time, and then a drift from that state corrected.
+    # This setup manages a file and then drifts its state. The noop added to the
+    # resource will still mark the report as noop_pending.
+    set_sitepp_content(to_manifest(declare('file', '/tmp/test', 'content' => 'hello')))
+    trigger_puppet_run(master)
```

Comment:
Hmm, could you `apply_manifest` this part? Would be faster than a full-blown server run.

I'm gonna guess you already tried that and it doesn't work with corrective changes b/c detecting a corrective change depends on info from a prior run, which `puppet apply` doesn't store? If that is the reason, I'd mention that in the comments because I'm sure someone else would look at this and wonder why we're not `apply_manifest`'ing there.

Reply:
Yeah I already tested this. Without a full puppet run no history is kept of what's been managed before, so no changes can be considered corrective.

Reply:
Ok, then I suggest updating the comment to mention why you're not using `apply_manifest` here because that's definitely a subtle detail.

Reply:
Done.

Reply:
`managed in the past` => `managed in the past (where the past history is used to calculate corrective changes)`

Reply:
Done